### PR TITLE
Fix limit and offset for tokens routes

### LINKED ISSUE #186: Sweep limit, offset parameters for all endpoints
The implementation of limit and offset parameters for all endpoints needs to be checked and corrected in several ways.

(1) Some endpoints have in memory limit,offset logic implemented, instead limit and offset should use the build in LIMIT and OFFSET functionalities of SQL, using knex.

(2) All endpoints should have min (0) and max (1000) parameters for limit, and a min parameter of 0 for offset.  Offset should default to 0, and limit should default to the max of 1000.

(3) Additionally the following issues need to be checked and resolved if work is still needed.

https://github.com/Greenstand/treetracker-wallet-api/issues/185
https://github.com/Greenstand/treetracker-wallet-api/issues/183
https://github.com/Greenstand/treetracker-wallet-api/issues/87

### PR DESCRIPTION
**Pull request by zo-chu**
partially fixes #186 

- Now tokens router is using offset and limit with the DB and not in code
- changed defaults and min/max according to ticket
- edited integration tests

### CODE REVIEW

File: __tests__/get-tokens.spec.js [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
+const chai = require('chai');
 chai.use(require('chai-uuid'));
 
 describe('GET tokens', () => {
   let bearerToken;
   let bearerTokenB;
 
-  before( async () => {
-
+  beforeEach(async () => {
```
Comment (dadiorchen) [inline __tests__/get-tokens.spec.js:17]: Yes, I think in the same way, to make the tests independent, and I also are building the test in the way similar with your wat to add some `seed` or data-preparing function to create the data senerioir, take a look this PR:
https://github.com/Greenstand/treetracker-wallet-api/pull/192

**Review by dadiorchen — APPROVED** [review]

**Comment by dadiorchen** [comment]
The integration tests now are not good, part of the reason is that we are copy these tests from previous tests which you can call them end to end test, we are planning to refactor them to a better way.

File: server/routes/tokenRouter.js [hunk 2]
```diff
@@ -53,19 +53,18 @@ tokenRouter.get('/',
     const walletService = new WalletService(session);
     const walletLogin = await walletService.getById(res.locals.wallet_id);
     let tokens = [];
+    
     if(wallet){
       const walletInstance = await walletService.getByName(wallet);
       const isSub = await walletLogin.hasControlOver(walletInstance);
       if(!isSub){
-        throw new HttpError(403, "Wallet do not belongs to wallet logged in");
+        throw new HttpError(403, "Wallet does not belong to wallet logged in");
```
[the remark this answers is not in the document]
Comment (zo-chu) [reply server/routes/tokenRouter.js:61]: haha, no worries

File: __tests__/get-tokens.spec.js (hunk 1 above)
Comment (zo-chu) [reply __tests__/get-tokens.spec.js:17]: yeah, I saw in some places tests were depending on each other and it's a bad practice - makes making changes very complicated and generally it's error prone. Ideally, every unit test should create data for itself and each time data should be erased after the test, I think

**Pull request merged**